magma/generator.py

```diff
@@ -104,6 +111,10 @@ def __new__(metacls, name, bases, dct):
     def __call__(cls, *args, **kwargs):
         return DefineCircuitKind.__call__(cls, *args, **kwargs)
 
+    @classmethod
+    def bind(cls, monitor):
```

Comment:
im not sure im understanding how this works. is there a simpler example test you can write (can be dumb simple). I'm just not seeing where this classmethod gets called.

Reply:
The reason you don't see it in the changes is because this simply updates old code using m.Generator to use m.Generator2, so the test code that calls it didn't change.  The bind call can be found here: https://github.com/phanrahan/magma/blob/master/tests/test_verilog/rtl_monitor.py#L46

We bind a monitor generator to a magma generator, so every time a generator is invoked to create a circuit, we call the monitor generator to create a monitor for that instance.

Reply:
The old m.Generator code can be found here:
* https://github.com/phanrahan/magma/blob/master/magma/generator.py#L20-L21
* https://github.com/phanrahan/magma/blob/master/magma/generator.py#L51-L53